lib/transports/odbcTransport.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) International Business Machines Corp. 2019
+// All Rights Reserved
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+// associated documentation files (the "Software"), to deal in the Software without restriction,
+// including without limitation the rights to use, copy, modify, merge, publish, distribute,
+// sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+// NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+function odbcCall(config, xmlInput, done) {
+  // eslint-disable-next-line global-require
+  const { Connection } = require('odbc');
+
+  const {
+    host = 'localhost',
+    username = null,
+    password = null,
+    ipc = '*NA',
+    ctl = '*here',
+    xslib = 'QXMLSERV',
+    verbose = false,
+    dsn = null,
+  } = config;
+
+  const sql = `call ${xslib}.iPLUGR512K(?,?,?)`;
+  const driver = 'IBM i Access ODBC Driver';
+  let connectionString;
+
+  if (dsn && typeof dsn === 'string') {
+    connectionString = `DSN=${dsn}`;
+  } else {
+    connectionString = `DRIVER=${driver};SYSTEM=${host};UID=${username};PWD=${password};`;
```

Comment:
If the user does not pass in a username/password, does this not evaluate to `UID=null;PWD=null`?

I'd suggest we guard on username/password being strings and only add them to the connection string if so:

```javascript
if (typeof username == 'string') {
    connectionString += `UID=${username};`;
}
if (typeof password == 'string') {
    connectionString += `PWD=${password};`;
}

Reply:
@kadler 

True in that case it will evaluate to `UID=null;PWD=null`.

Adding on the the above suggestion, we should probably also guard against the empty string case:

```
if (username && typeof username == 'string') {
    connectionString += `UID=${username};`;
}
if (username && typeof password == 'string') {
    connectionString += `PWD=${password};`;
}
```